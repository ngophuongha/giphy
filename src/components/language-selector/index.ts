export * from "./LanguageSelector.styled";
export * from "./LanguageSelector";
