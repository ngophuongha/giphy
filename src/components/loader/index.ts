export * from "./Loader";
export * from "./Suspender";
