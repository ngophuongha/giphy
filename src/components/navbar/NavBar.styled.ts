import styled from "styled-components";
import { screens } from "../../themes";

export const StyledNav = styled.a`
  color: ${({ theme }) => theme.colors.textColor};
  text-decoration: none;
  padding: 0.5rem 1rem;
  display: block;
  &:hover {
    color: ${({ theme }) => theme.colors.textColor};
  }
  @media only screen and (max-width: ${screens.lg}) {
    a {
      padding-left: 0.5rem;
      padding-right: 0.5rem;
    }
  }
`;
