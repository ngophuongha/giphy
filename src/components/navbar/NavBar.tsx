import { Fragment } from "react";
import { useTranslation } from "react-i18next";
import { Nav, Navbar, Container } from "react-bootstrap";

import { ThemeToggler } from "../theme-toggler";
import { StyledNav } from "./NavBar.styled";

export const NavBar = (): JSX.Element => {
  const { t } = useTranslation("common");
  return (
    <Fragment>
      <Navbar
        collapseOnSelect
        expand="lg"
        variant="light"
        className="border-bottom"
      >
        <Container>
          <StyledNav href="#home">
            <img
              alt=""
              src="/giphy-logo-1.svg"
              width="30"
              height="30"
              className="d-inline-block align-top"
            />{" "}
            GIPHY
          </StyledNav>
          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
          <Navbar.Collapse id="responsive-navbar-nav">
            <Nav className="me-auto">
              <StyledNav href="#">{t("navbar.reactions")}</StyledNav>
              <StyledNav href="#">{t("navbar.entertainment")}</StyledNav>
            </Nav>
            <Nav className="flex-row align-items-center">
              <ThemeToggler />
              <StyledNav href="#">{t("navbar.signIn")}</StyledNav>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </Fragment>
  );
};
