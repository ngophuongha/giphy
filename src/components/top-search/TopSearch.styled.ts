import styled from "styled-components";
import { Button, FormControl } from "react-bootstrap";

export const StyledFormControl = styled(FormControl)`
  background-color: ${({ theme }) => theme.colors.background} !important;
  color: ${({ theme }) => theme.colors.textColor} !important;
`;

export const StyledSearchBtn = styled(Button)`
  background-color: ${({ theme }) => theme.colors.background} !important;
  color: ${({ theme }) => theme.colors.textColor} !important;
  border-color: ${({ theme }) => theme.colors.borderColor} !important;
  &:hover {
    background-color: ${({ theme }) => theme.colors.hoverBackground} !important;
    color: ${({ theme }) => theme.colors.hoverTextColor} !important;
  }
`;
