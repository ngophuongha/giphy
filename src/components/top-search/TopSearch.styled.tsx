import styled from "styled-components";
import { FormControl } from "react-bootstrap";

export const StyledFormControl = styled(FormControl)`
  background-color: ${({ theme }) => theme.colors.background} !important;
  color: ${({ theme }) => theme.colors.textColor} !important;
`;
