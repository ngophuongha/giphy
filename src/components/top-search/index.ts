export * from "./TopSearch.styled";
export * from "./TopSearch";
