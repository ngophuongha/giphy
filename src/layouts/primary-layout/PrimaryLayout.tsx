import { Container } from "react-bootstrap";
import { NavBar } from "../../components/navbar";
import { TopSearch } from "../../components/top-search";
import { Wrapper } from "../../components/wrapper";

import styles from "./PrimaryLayout.module.scss";

export const PrimaryLayout: React.FC = ({ children }) => {
  return (
    <Wrapper>
      <Container className={styles.container}>
        <NavBar />
        <TopSearch />
        {children}
      </Container>
    </Wrapper>
  );
};
