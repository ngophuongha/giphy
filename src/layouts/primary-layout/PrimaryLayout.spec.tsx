import { Suspense } from "react";
import { render, waitFor, screen } from "../../../test";

import { PrimaryLayout } from "./PrimaryLayout";
import { NavBar } from "../../components/navbar";
import { TopSearch } from "../../components/top-search";

jest.mock("../../components/navbar", () => ({
  NavBar() {
    return <>NavBar</>;
  },
}));

jest.mock("../../components/top-search", () => ({
  TopSearch() {
    return <>Top Search</>;
  },
}));

describe("Primary Layout", () => {
  it("should match snapshot", async () => {
    const { container } = await render(
      <Suspense fallback={<></>}>
        <PrimaryLayout />
      </Suspense>
    );
    await waitFor(() => {
      expect(screen.getByText(/search/i)).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });

  it("should render children", async () => {
    render(
      <Suspense fallback={<></>}>
        <PrimaryLayout>
          <div>children</div>
        </PrimaryLayout>
      </Suspense>
    );
    await waitFor(() =>
      expect(screen.getByText(/children/i)).toBeInTheDocument()
    );
  });
});
