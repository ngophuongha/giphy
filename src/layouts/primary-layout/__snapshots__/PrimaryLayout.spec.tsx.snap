// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Layout should match snapshot 1`] = `
.c1 {
  background-color: #fff !important;
  color: #000000 !important;
}

.c2 {
  border: 1px solid #d3d3d3;
  position: inherit;
}

.c3 {
  padding: 0.25rem !important;
  border-radius: 0.25rem;
  border: 1px solid #d3d3d3;
  color: #000000;
  background-color: #fff;
}

.c3:hover {
  background-color: #444444;
  color: #ffffff;
}

.c0 {
  background-color: #fff;
  color: #000000;
  overflow-x: hidden;
}

<div>
  <div
    class="c0"
  >
    <div
      class="container container"
    >
      NavBar
      <div
        class="d-flex w-100"
      >
        <div
          class="d-flex w-100"
        >
          <div
            class="my-4 input-group"
          >
            <div
              class="d-flex w-100 flex-column"
            >
              <input
                aria-label="gif-sticker-search"
                class="c1 form-control"
                placeholder="searchPlaceholder"
                value=""
              />
              <div
                class="c2 dropdown-menu"
                x-placement="bottom-start"
              />
            </div>
          </div>
        </div>
        <div>
          <button
            class="c3 mt-4"
            id="search-btn"
          >
            searchBtn
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
